File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.py
def solution(n, words):
    answer = [0,0]
    memory = []
    
    for i in range(len(words)-1):
        if words[i][-1] != words[i+1][0] or words[i] in memory or words[i+1] in memory:
            answer[0] = (i+1) % n + 1
            answer[1] = (i+1) // n + 1
            break
        else :
            memory.append(words[i])

    return answer

File: 프로그래머스/2/12981. 영어 끝말잇기/영어 끝말잇기.java
import java.util.*;

class Solution {
    public int[] solution(int n, String[] words) {
        int[] answer = new int[2];
        ArrayList<String> used = new ArrayList<>();
        
        char lastChar = words[0].charAt(words[0].length()-1);
        used.add(words[0]);
        
        for(int i = 1; i<words.length; i++){
            char tmp = words[i].charAt(0);
            if(tmp == lastChar && !used.contains(words[i])){
                lastChar = words[i].charAt(words[i].length()-1);
                used.add(words[i]);
            }
            else{
                answer[0] = (i%n) + 1;
                answer[1] = ((i - (i%n)) / n)+1;
                break;
            }
        }

        return answer;
    }
}